pkg/controller/job/job_controller_test.go

```diff
@@ -3040,6 +3040,76 @@ func TestSyncJobWithJobPodFailurePolicy(t *testing.T) {
 			},
 			wantStatusActive: 1, // This is a replacement Pod: the terminating Pod is neither active nor failed.
 		},
+		"Terminating and Running Pod test case with delay pod flag set to true": {
```

Comment:
IIUC, the two new test cases don't show any difference in `wantStatusActive` and `wantStatusFailed`. I think the minimal scenario should be one terminating pod. When the new field is enabled, then the number of active is 1, otherwise 0?

Reply:
> Do you know how the state transitions work in these test cases?

Not very familiar, I would need to take a look. I could take a look tomorrow probably if you are stuck on this.